### PR TITLE
Add support for Redis as key storage

### PR DESCRIPTION
**Pull request by ndreas**
This adds a very slim implementation for using Redis as the storage of idempotency keys. The interface already matched the key-value nature of Redis quite well, but I extended it to use `context.Context`.

I also opted to use `SETNX` for the actual setting of data. There is a tiny possibility for a race condition if you perform a `GET` followed by a `SET` in a multi process environment. You could in theory issue the `GET`, receive that it's not set, and `SET` while another process managed a `SET` in between. Using `SETNX` we can handle this case.

### CODE REVIEW

File: storage.go [hunk 1]
```diff
@@ -44,11 +52,68 @@ func (m *memoryStorage) Get(key string) (*RequestStatus, error) {
...
 
 	return nil
 }
+
+type redisStorage struct {
+	client *redis.Client
+	expiry time.Duration
+}
+
+const keyPrefix = "idemp:"
```
Comment (preciselytord) [inline storage.go]: i think we need to handle the prefixing in the same way as in the worker queue to handle the case where we share the redis instance in staging.

util.KubernetesNSSuffix(pwork.NAMESPACE, config.Env.PodNamespace)

**Review by preciselytord — CHANGES_REQUESTED** [review]
otherwise it looks good

File: storage.go (hunk 1 above)
Comment (ndreas) [reply storage.go]: Good catch, I will make it configurable here since this is a generic package.

**Review by preciselytord — APPROVED** [review]

**Review by AdamMK — COMMENTED** [review]
LGTM

**Pull request merged**